planning/behavior_path_planner/autoware_behavior_path_goal_planner_module/src/goal_searcher.cpp

```diff
@@ -193,8 +196,23 @@ GoalCandidates GoalSearcher::search(const std::shared_ptr<const PlannerData> & p
         continue;
       }
 
+      // modify the goal_pose orientation so that vehicle footprint front heading is parallel to the
+      // lane boundary
+      const auto vehicle_front_midpoint =
+        (transformed_vehicle_footprint.at(vehicle_info_utils::VehicleInfo::FrontLeftIndex) +
+         transformed_vehicle_footprint.at(vehicle_info_utils::VehicleInfo::FrontRightIndex)) /
+        2.0;
+      lanelet::ConstLanelet vehicle_front_closest_lanelet;
+      lanelet::utils::query::getClosestLanelet(
+        pull_over_lanes, search_pose, &vehicle_front_closest_lanelet);
+      const auto vehicle_front_pose_for_bound = goal_planner_utils::calcClosestPose(
+        left_side_parking_ ? vehicle_front_closest_lanelet.leftBound()
+                           : vehicle_front_closest_lanelet.rightBound(),
+        autoware::universe_utils::createPoint(
+          vehicle_front_midpoint.x(), vehicle_front_midpoint.y(), search_pose.position.z));
```

Comment:
```suggestion
      const auto vehicle_front_pose_for_bound = goal_planner_utils::calcClosestPose(
        left_side_parking_ ? vehicle_front_closest_lanelet.leftBound()
                           : vehicle_front_closest_lanelet.rightBound(),
        autoware::universe_utils::createPoint(
          vehicle_front_midpoint.x(), vehicle_front_midpoint.y(), search_pose.position.z));
      if (!vehicle_front_pose_for_bound.has_value()) continue;
```

or set search_pose.orientation is better?